packages/client/hmi-client/src/components/widgets/chatty-input.vue

```diff
@@ -0,0 +1,200 @@
+<template>
+	<div class="chatty-container" ref="containerElement">
+		<label for="chatt-input"></label>
+		<input
+			ref="inputElement"
+			id="chatty-input"
+			v-model="queryString"
+			type="text"
+			:disabled="kernelState !== KernelState.idle"
+			:style="inputStyle"
+			:placeholder="placeholderMessage"
+			@keydown.enter="onEnter"
+		/>
+	</div>
+	<div id="chatty-history">
+		<div v-for="message in messages" :key="message.msg_id">
+			<div v-if="message.msg_type === 'execute_input'" class="query">
+				Query: {{ message.content?.code }}
+			</div>
+			<div
+				v-else-if="
+					message.msg_type === 'chatty_response' && message.content.name === 'response_text'
+				"
+				class="answer"
+			>
+				Response: {{ message.content?.text }}
+			</div>
+			<div
+				v-else-if="message.msg_type === 'stream' && message.content.name === 'stdout'"
+				class="thought"
+			>
+				{{ message.content?.text }}
+			</div>
+			<div
+				v-else-if="message.msg_type === 'stream' && message.content.name === 'stderr'"
+				class="error"
+			>
+				Error: {{ message.content?.text }}
+			</div>
+			<div v-else>Other: {{ message }}</div>
+			<hr />
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+// @ts-nocheck
+import { computed, onMounted, ref } from 'vue';
+
+import { SessionContext } from '@jupyterlab/apputils';
+import {
+	ServerConnection,
+	KernelManager,
+	KernelSpecManager,
+	SessionManager
+} from '@jupyterlab/services';
+import { IIOPubMessage } from '@jupyterlab/services/lib/kernel/messages';
+
+// TODO: These settings should be pulled from the environment variables or appropriate config setup.
+const serverSettings = ServerConnection.makeSettings({
```

Comment:
The way we've done this for similar configuration/secrets needed on the front end is to have an endpoint in HMI server that returns the appropriate config.

https://github.com/DARPA-ASKEM/Terarium/pull/1029 is an example.